qutebrowser/misc/sql.py

```diff
@@ -469,9 +533,19 @@ def select(self, sort_by, sort_order, limit=-1):
 
         Return: A prepared and executed select query.
         """
-        q = Query("SELECT * FROM {table} ORDER BY {sort_by} {sort_order} "
-                  "LIMIT :limit"
-                  .format(table=self._name, sort_by=sort_by,
-                          sort_order=sort_order))
+        q = self._database.query(
+            f"SELECT * FROM {self._name} ORDER BY {sort_by} {sort_order} LIMIT :limit"
+        )
         q.run(limit=limit)
         return q
+
+
+def version() -> str:
+    """Return the sqlite version string."""
+    try:
+        in_memory_db = Database(':memory:')
+        version = in_memory_db.query("select sqlite_version()").run().value()
+        in_memory_db.close()
```

Comment:
You could use `with contextlib.closing(Database(':memory:')) as db:` here, so that `.close()` gets called properly even if the query fails for some reason. See [docs](https://docs.python.org/3/library/contextlib.html#contextlib.closing).

Reply:
Aha, nice one. Thanks.